internal/configuration/schema/access_control.go

```diff
@@ -1,26 +1,31 @@
 package schema
 
+import (
+	"regexp"
+)
+
 // AccessControlConfiguration represents the configuration related to ACLs.
 type AccessControlConfiguration struct {
 	DefaultPolicy string       `koanf:"default_policy"`
 	Networks      []ACLNetwork `koanf:"networks"`
 	Rules         []ACLRule    `koanf:"rules"`
 }
 
-// ACLNetwork represents one ACL network group entry; "weak" coerces a single value into slice.
+// ACLNetwork represents one ACL network group entry.
 type ACLNetwork struct {
 	Name     string   `koanf:"name"`
 	Networks []string `koanf:"networks"`
 }
 
-// ACLRule represents one ACL rule entry; "weak" coerces a single value into slice.
+// ACLRule represents one ACL rule entry.
 type ACLRule struct {
-	Domains   []string   `koanf:"domain"`
-	Policy    string     `koanf:"policy"`
-	Subjects  [][]string `koanf:"subject"`
-	Networks  []string   `koanf:"networks"`
-	Resources []string   `koanf:"resources"`
-	Methods   []string   `koanf:"methods"`
+	Domains      []string         `koanf:"domain"`
+	DomainsRegex []*regexp.Regexp `koanf:"domain_regex"`
```

Comment:
We could deref if you think it's a good idea?

Reply:
the earliest the nil pointer check is done, the better. From my experience with C/C++, if you can avoid pointers, you should. This is to avoid nil pointers troubles and avoid having nil checks in every function, that's why I'm asking if we can do without it.